src/lib/fetch.ts

```diff
@@ -91,11 +102,42 @@ export async function azleFetch(input: any, init?: any): Promise<any> {
             text: async () => {
                 return contents.toString();
             }
-        };
+        } as any;
     }
 
-    if (url.protocol === 'http:') {
-        // TODO http call to the management canister
+    if (url.protocol === 'https:' || url.protocol === 'http:') {
+        const body = await prepareRequestBody(init);
+        // TODO also do headers and method and everything like on the client?
+
```

Comment:
There are also a number of constraints on the url and headers. Do we need to enforce those? Or would us enforcing them be an opportunity to have a better error message for our users?
- 8192 url max
- No IPv4 addresses
- 64 headers
- 8 KiB limit on header names and values
- 48 KiB limit on all total header names and values

Reply:
We've decided it would be best just let the IC enforce this, but it would be good to have this in the documentation. Please make an issue to address this when we do documentation

Reply:
I have added a checkbox for this here: https://github.com/demergent-labs/azle/issues/1698